src/simulation.jl

```diff
@@ -164,18 +181,24 @@ function compile_env(d::AbstractNamedDecapode, dec_matrices::Vector{Symbol}, con
     quote_op = QuoteNode(op)
     mat_op = add_stub(gensim_in_place_stub, op)
     # Could turn this into a special call
-    def = :(($mat_op, $op) = default_dec_matrix_generate(mesh, $quote_op, hodge))
+
+    default_generation = :default_dec_matrix_generate
```

Comment:
Use ternary operator

Reply:
Can't since we could be using either in-place or out-of-place in the same simulation. Also this would break the way default_dec_matrix_generate works.

Reply:
Oh, I didn't realize you meant choosing the default generation. I've updated this to an MLStyle in-case we want to add more targets. Also I've TODOed this line so when we allow users to overload hooks with their own code targets, we can update this.